lib/index.js

```diff
@@ -218,11 +218,26 @@ const parseFragment = (fragment, name = []) => {
 	return result
 }
 
-module.exports = (database, table, schema) => {
-	const query = rethinkdb
+const useSecondaryIndexes = (query, schema, options = {}) => {
+	if(options.indexes !== undefined) {
+		const schemaProperties = schema.properties || {}
+		for(const index of Object.keys(options.indexes)) {
+			idxQuery = schemaProperties[index] || {}
+			if(idxQuery.const !== undefined) {
+				query = query.getAll(idxQuery.const, {index})
+			} else if(idxQuery.enum !== undefined) {
+				query = query.getAll(idxQuery.enum, {index})
```

Comment:
When I tested this locally, it seems that if there is `enum` then the query won't return a value.
With a new index `type` created, the following query returns no results:
```
r.db('jellyfish').table('cards').getAll([ 'user' ], {index: 'type'})
```
whereas this query does:
```
r.db('jellyfish').table('cards').getAll('user', {index: 'type'})
```
Close investigation shows that you can't pass an array as the first argument to `.getAll()` and instead you need to pass multiple arguments. Here you can achieve this using a spread operator:
```
query = query.getAll(...idxQuery.enum, {index})
```